xarray/core/common.py

```diff
@@ -381,11 +382,11 @@ class DataWithCoords(AttrAccessMixin):
     __slots__ = ("_close",)
 
     def squeeze(
-        self: T_DataWithCoords,
+        self: Self,
```

Comment:
```suggestion
        self,
```
This Self is unnecessary